### ISSUE TITLE
Initialization inside a loop

### ISSUE THREAD

**Issue by MamadouSDiallo** [astral-sh/ty#1014]
### Summary

I can initialize the variable before the loop but I expected this code to pass as is.

```py
def test():
    
    for i in range(10):
        if i==0:
            var = 1
        else:
            var = var + 1
```

```py
error[unresolved-reference]: Name `var` used when not defined
 --> example2.py:7:19
  |
5 |             var = 1
6 |         else:
7 |             var = var + 1
  |                   ^^^
  |
info: rule `unresolved-reference` is enabled by default
```

**Comment by carljm** [comment]
This should be changed from `unresolved-reference` to `possibly-unresolved-reference` (which is disabled by default) by #232 .

Note that it's kind of dangerous to allow this code to pass without error; mypy doesn't actually understand that it will definitely be defined on the first iteration of the loop, it's just trusting you to get it right. (You can see this by changing `i == 0` to `i == 999` and mypy is still fine with it.)
